com.intellij.diagnostic.PluginException: Cannot init component state

Rider 2021.1-EAP on Windows

```
com.intellij.diagnostic.PluginException: Cannot init component state (componentName=net.cakebuild.settings.CakeSettings, componentClass=CakeSettings) [Plugin: net.cakebuild.cakerider]
	at com.intellij.configurationStore.ComponentStoreImpl.initComponent(ComponentStoreImpl.kt:144)
	at com.intellij.configurationStore.ComponentStoreWithExtraComponents.initComponent(ComponentStoreWithExtraComponents.kt:49)
	at com.intellij.serviceContainer.ComponentManagerImpl.initializeComponent$intellij_platform_serviceContainer(ComponentManagerImpl.kt:433)
	at com.intellij.serviceContainer.ServiceComponentAdapter.createAndInitialize(ServiceComponentAdapter.kt:53)
	at com.intellij.serviceContainer.ServiceComponentAdapter.access$createAndInitialize(ServiceComponentAdapter.kt:13)
	at com.intellij.serviceContainer.ServiceComponentAdapter$doCreateInstance$1.run(ServiceComponentAdapter.kt:43)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:688)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:634)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:64)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeNonCancelableSection(CoreProgressManager.java:218)
	at com.intellij.serviceContainer.ServiceComponentAdapter.doCreateInstance(ServiceComponentAdapter.kt:42)
	at com.intellij.serviceContainer.BaseComponentAdapter.getInstanceUncached(BaseComponentAdapter.kt:110)
	at com.intellij.serviceContainer.BaseComponentAdapter.getInstance(BaseComponentAdapter.kt:64)
	at com.intellij.serviceContainer.ComponentManagerImpl.doGetService(ComponentManagerImpl.kt:488)
	at com.intellij.serviceContainer.ComponentManagerImpl.getService(ComponentManagerImpl.kt:471)
	at com.intellij.openapi.components.ServiceManager.getService(ServiceManager.java:26)
	at net.cakebuild.settings.CakeSettings$Companion.getInstance(CakeSettings.kt:19)
	at net.cakebuild.fileTypes.CakeFileType.isMyFileType(CakeFileType.kt:25)
	at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.getByFile(FileTypeManagerImpl.java:626)
	at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.getFileTypeByFile(FileTypeManagerImpl.java:584)
	at com.intellij.openapi.fileTypes.impl.FileTypeManagerImpl.getFileTypeByFile(FileTypeManagerImpl.java:573)
	at com.intellij.openapi.vfs.VirtualFile.getFileType(VirtualFile.java:354)
	at com.intellij.openapi.vfs.newvfs.impl.VirtualFileSystemEntry.getFileType(VirtualFileSystemEntry.java:578)
	at com.intellij.openapi.fileEditor.impl.FileDocumentManagerBase.isBinaryWithoutDecompiler(FileDocumentManagerBase.java:134)
	at com.intellij.openapi.fileEditor.impl.FileDocumentManagerBase.getDocument(FileDocumentManagerBase.java:41)
	at com.intellij.openapi.fileEditor.impl.EditorsSplitters$UIBuilder.lambda$processFiles$1(EditorsSplitters.java:971)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:848)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:61)
	at com.intellij.openapi.fileEditor.impl.EditorsSplitters$UIBuilder.processFiles(EditorsSplitters.java:970)
	at com.intellij.openapi.fileEditor.impl.EditorsSplitters$UIBuilder.processFiles(EditorsSplitters.java:933)
	at com.intellij.openapi.fileEditor.impl.EditorsSplitters$ConfigTreeReader.process(EditorsSplitters.java:925)
	at com.intellij.openapi.fileEditor.impl.EditorsSplitters.restoreEditors(EditorsSplitters.java:266)
	at com.intellij.openapi.fileEditor.impl.OpenFilesActivity.runActivity(OpenFilesActivity.java:36)
	at com.intellij.ide.startup.impl.StartupManagerImpl.runStartupActivity(StartupManagerImpl.java:335)
	at com.intellij.ide.startup.impl.StartupManagerImpl.runActivity(StartupManagerImpl.java:309)
	at com.intellij.ide.startup.impl.StartupManagerImpl.lambda$executeActivitiesFromExtensionPoint$2(StartupManagerImpl.java:207)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processImplementations(ExtensionPointImpl.java:320)
	at com.intellij.ide.startup.impl.StartupManagerImpl.executeActivitiesFromExtensionPoint(StartupManagerImpl.java:188)
	at com.intellij.ide.startup.impl.StartupManagerImpl.runStartUpActivities(StartupManagerImpl.java:180)
	at com.intellij.ide.startup.impl.StartupManagerImpl.projectOpened(StartupManagerImpl.java:145)
	at com.intellij.openapi.project.impl.ProjectManagerExImplKt.openProject(ProjectManagerExImpl.kt:387)
	at com.intellij.openapi.project.impl.ProjectManagerExImplKt.access$openProject(ProjectManagerExImpl.kt:1)
	at com.intellij.openapi.project.impl.ProjectManagerExImpl.doOpenProject(ProjectManagerExImpl.kt:176)
	at com.intellij.openapi.project.impl.ProjectManagerExImpl.access$doOpenProject(ProjectManagerExImpl.kt:55)
	at com.intellij.openapi.project.impl.ProjectManagerExImpl$openProjectAsync$1.invoke(ProjectManagerExImpl.kt:119)
	at com.intellij.openapi.project.impl.ProjectManagerExImpl$openProjectAsync$1.invoke(ProjectManagerExImpl.kt:55)
	at com.intellij.openapi.project.impl.ProjectUiFrameAllocator$run$progressRunner$1.apply(ProjectFrameAllocator.kt:92)
	at com.intellij.openapi.project.impl.ProjectUiFrameAllocator$run$progressRunner$1.apply(ProjectFrameAllocator.kt:69)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$3(ProgressRunner.java:228)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:178)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:688)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:634)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:64)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:165)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:228)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: com.intellij.util.xmlb.XmlSerializationException: Cannot deserialize class net.cakebuild.settings.CakeSettings
	at com.intellij.configurationStore.JdomSerializerImpl.deserialize(xmlSerializer.kt:96)
	at com.intellij.configurationStore.XmlSerializer.deserialize(xmlSerializer.kt:24)
	at com.intellij.configurationStore.DefaultStateSerializerKt.deserializeState(DefaultStateSerializer.kt:26)
	at com.intellij.configurationStore.StateStorageBase.deserializeState(StateStorageBase.kt:35)
	at com.intellij.configurationStore.StateGetterImpl.getState(StorageBaseEx.kt:57)
	at com.intellij.configurationStore.ComponentStoreImpl.doInitComponent(ComponentStoreImpl.kt:422)
	at com.intellij.configurationStore.ComponentStoreImpl.initComponent(ComponentStoreImpl.kt:372)
	at com.intellij.configurationStore.ComponentStoreImpl.initComponent(ComponentStoreImpl.kt:123)
	... 62 more
Caused by: java.lang.UnsupportedOperationException
	at java.base/java.util.Collections$1.remove(Collections.java:4714)
	at java.base/java.util.AbstractCollection.clear(AbstractCollection.java:447)
	at java.base/java.util.AbstractMap.clear(AbstractMap.java:297)
	at com.intellij.util.xmlb.MapBinding.deserialize(MapBinding.java:182)
	at com.intellij.util.xmlb.MapBinding.deserializeList(MapBinding.java:160)
	at com.intellij.util.xmlb.Binding.deserializeList(Binding.java:28)
	at com.intellij.util.xmlb.OptionTagBinding.deserialize(OptionTagBinding.java:107)
	at com.intellij.util.xmlb.BasePrimitiveBinding.deserializeUnsafe(BasePrimitiveBinding.java:54)
	at com.intellij.util.xmlb.BeanBinding.deserializeInto(BeanBinding.java:238)
	at com.intellij.util.xmlb.BeanBinding.deserializeInto(BeanBinding.java:192)
	at com.intellij.util.xmlb.BeanBinding.deserialize(BeanBinding.java:142)
	at com.intellij.configurationStore.JdomSerializerImpl.deserialize(xmlSerializer.kt:90)
	... 69 more
```

--- a/rider/src/main/kotlin/net/cakebuild/settings/CakeSettings.kt
+++ b/rider/src/main/kotlin/net/cakebuild/settings/CakeSettings.kt
@@ -26,7 +26,7 @@ class CakeSettings : PersistentStateComponent<CakeSettings> {
     var cakeTaskParsingRegex = "Task\\s*?\\(\\s*?\"(.*?)\"\\s*?\\)"
     var cakeVerbosity = "normal"
     var cakeRunner = "~/.dotnet/tools/dotnet-cake"
-    var cakeRunnerOverrides = mapOf(Pair("^.*windows.*$", "\${USERPROFILE}\\.dotnet\\tools\\dotnet-cake.exe"))
+    var cakeRunnerOverrides = mutableMapOf(Pair("^.*windows.*$", "\${USERPROFILE}\\.dotnet\\tools\\dotnet-cake.exe"))
     var cakeScriptSearchPaths: Collection<String> = mutableListOf(".")
     var cakeScriptSearchIgnores: Collection<String> = mutableListOf(".*/tools/.*")
     // do we need a settings-UI for these?

